pkg/apis/certificates/validation/validation.go

```diff
@@ -537,3 +537,118 @@ func hasDuplicateUsage(usages []certificates.KeyUsage) bool {
 	}
 	return false
 }
+
+func requireSignerNameAsPrefix(signerName string) func(name string, prefix bool) []string {
+	return func(name string, prefix bool) []string {
+		if signerName != "" {
+			escaped := strings.ReplaceAll(signerName, ".", "-")
+			escaped = strings.ReplaceAll(escaped, "/", "--")
+			if !strings.HasPrefix(name, escaped) {
+				return []string{
+					fmt.Sprintf("ClusterTrustBundle for signer %s must be named with prefix %s", signerName, escaped),
+				}
+			}
+		}
+
+		return nil
+	}
+}
+
+// ValidateClusterTrustBundle runs all validation checks on bundle.
+func ValidateClusterTrustBundle(bundle *certificates.ClusterTrustBundle) field.ErrorList {
+	var allErrors field.ErrorList
+
+	metaErrors := apivalidation.ValidateObjectMeta(&bundle.ObjectMeta, false, requireSignerNameAsPrefix(bundle.Spec.SignerName), field.NewPath("metadata"))
+	allErrors = append(allErrors, metaErrors...)
+
+	if bundle.Spec.SignerName != "" {
+		signerNameErrors := ValidateSignerName(field.NewPath("spec", "signerName"), bundle.Spec.SignerName)
+		allErrors = append(allErrors, signerNameErrors...)
+	}
+
+	pemErrors := validatePEMTrustAnchors(bundle.Spec.PEMTrustAnchors)
+	allErrors = append(allErrors, pemErrors...)
+
+	return allErrors
+}
+
+// ValidateClusterTrustBundleUpdate runs all update validation checks on an
+// update.
+func ValidateClusterTrustBundleUpdate(oldBundle, newBundle *certificates.ClusterTrustBundle) field.ErrorList {
+	var allErrors field.ErrorList
+	allErrors = append(allErrors, ValidateClusterTrustBundle(newBundle)...)
+	allErrors = append(allErrors, apivalidation.ValidateObjectMetaUpdate(&newBundle.ObjectMeta, &oldBundle.ObjectMeta, field.NewPath("metadata"))...)
+
+	if newBundle.Spec.SignerName != oldBundle.Spec.SignerName {
```

Comment:
use ValidateImmutableField() for this, and can drop the check from admission

Reply:
Done.